server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/DSNBounce.java

```diff
@@ -205,186 +176,154 @@ public void service(Mail originalMail) throws MessagingException {
-            // attribute we can use here
-            String ex = (String) originalMail.getAttribute("delivery-error");
+    private String getHostname() {
+        try {
+            InetAddress hostAddress = InetAddress.getLocalHost();
+            return hostAddress.getHostName();
+        } catch (Exception e) {
+            return "[address unknown]";
+        }
+    }
 
-            if (ex == null) {
-                ex = "unknown";
-            }
-            out.println("Status: " + ex);
-
-            // optional: remote MTA
-            // to which MTA were we talking while the Error occured?
-
-            // optional: diagnostic-code
-            String diagnosticType;
-            // this typically is the return value received during smtp
-            // (or other transport) communication
-            // and should be stored as attribute by the smtp handler
-            // but until now we only have error-messages.
-            String diagnosticCode = ex;
-            // Sometimes this is the smtp diagnostic code,
-            // but James often gives us other messages
-            boolean smtpDiagCodeAvailable = diagPattern.matcher(diagnosticCode).matches();
-            if (smtpDiagCodeAvailable) {
-                diagnosticType = "smtp";
-            } else {
-                diagnosticType = "X-James";
-            }
-            out.println("Diagnostic-Code: " + diagnosticType + "; " + diagnosticCode);
+    private String bounceMessage() {
+        if (messageString.contains(MACHINE_PATTERN)) {
+            return messageString.replace(MACHINE_PATTERN, getHostname());
+        }
+        return messageString;
+    }
+
+    private MimeBodyPart createDSN(Mail originalMail) throws MessagingException {
+        StringBuffer buffer = new StringBuffer();
+
+        appendReportingMTA(buffer);
+        buffer.append("Received-From-MTA: dns; " + originalMail.getRemoteHost())
+            .append(LINE_BREAK);
+
+        for (MailAddress rec : originalMail.getRecipients()) {
+            appendRecipient(buffer, rec, getDeliveryError(originalMail), originalMail.getLastUpdated());
+        }
 
-            // optional: last attempt
-            out.println("Last-Attempt-Date: " + DateFormats.RFC822_DATE_FORMAT.format(originalMail.getLastUpdated()));
+        MimeBodyPart bodyPart = new MimeBodyPart();
+        bodyPart.setContent(buffer.toString(), "text/plain");
+        bodyPart.setHeader("Content-Type", "message/delivery-status");
+        bodyPart.setDescription("Delivery Status Notification");
+        bodyPart.setFileName("status.dat");
+        return bodyPart;
+    }
 
-            // optional: retry until
-            // only for 'delayed' reports .. but we don't report this (at least
-            // until now)
+    private void appendReportingMTA(StringBuffer buffer) {
+        try {
+            buffer.append("Reporting-MTA: dns; " + dns.getHostName(dns.getLocalHost()))
```

Comment:
Maybe not concerned by your review but can't this report an unrelated domain name ?